### ISSUE TITLE
Try demo with my own photo

### ISSUE THREAD

**Issue by JordanPeltier** [dougsouza/face-frontalization#5]
Hi and thank you for this work. 
I tried demo.py on my computer and it works well. 
The I tried demo.py with a picture of myself (taken with my webcam) and the result is totally wrong. 
Any ideas ? :)

**Comment by dougsouza** [comment]
Hi @Justidan,

Well, it may happen. This code has a limit for face position it can frontalize. What I can suggest is that you plot the landmarks detected in your face, if the points lay in the correct position (eyes, nose, mouth, jaw) everything should work. You can also try different photos of different angles and occlusion to see what happens.

It is also important to note cases where there is more than one face in the image, it is necessary to frontalize on by one.

Hope it helps

**Comment by JordanPeltier** [comment]
The problem is that the facial landmarks are very good. 
And I tried with different angles.
Let me show you : 
![figure_1](https://cloud.githubusercontent.com/assets/13131858/15297254/fb4efa68-1b90-11e6-988b-c04e8bf40092.png)
![figure_2](https://cloud.githubusercontent.com/assets/13131858/15297256/fc54654c-1b90-11e6-8ec6-f59180cea630.png)
![figure_3](https://cloud.githubusercontent.com/assets/13131858/15297258/fd4e7582-1b90-11e6-904a-99c117cd8460.png)
![figure_4](https://cloud.githubusercontent.com/assets/13131858/15297261/fe336b2e-1b90-11e6-8fba-501ab7649df8.png)

**Comment by dougsouza** [comment]
That's really weird. Let me run some tests to see what is going on.

**Comment by dougsouza** [comment]
I have tested on your photo and it worked as expected. Which version of OpenCV are you using? Maybe there is some issue when calculating the camera matrix.

![figure_2](https://cloud.githubusercontent.com/assets/16470848/15297572/3b742e26-1b71-11e6-9c87-78d2be17cec6.png)
![figure_4](https://cloud.githubusercontent.com/assets/16470848/15297580/3fbf869c-1b71-11e6-965e-3070e5664574.png)

**Comment by JordanPeltier** [comment]
Hi, 
So it may be my set up : 

> > > cv2.**version**
> > > '3.1.0'
> > > dlib.**version**
> > > '18.18.100'

**Comment by dougsouza** [comment]
Maybe it is. Dlib 18.18 is known to work, but to be honest, I haven't tested on OpenCV 3.1, only on 2.4 and 3.0.

**Comment by JordanPeltier** [comment]
Ok thanks for this reply !

**Comment by dougsouza** [comment]
@Justidan,

I experienced an issue similar to yours. It happened because the face was small compared to the image size. I still don't know why this happens, hope to investigate this soon.

**Comment by JordanPeltier** [comment]
@dougsouza 
Thanks for considering my issue. 
I attach what i get with bigger face.
![figure_1](https://cloud.githubusercontent.com/assets/13131858/15354017/95b1040e-1ce3-11e6-8273-ff555272c7cc.png)
![figure_2](https://cloud.githubusercontent.com/assets/13131858/15354019/95b481b0-1ce3-11e6-8d61-19dd8f66e9cd.png)
![figure_4](https://cloud.githubusercontent.com/assets/13131858/15354020/95b56f94-1ce3-11e6-94b7-b08fc8d54c72.png)
![figure_3](https://cloud.githubusercontent.com/assets/13131858/15354018/95b3cf2c-1ce3-11e6-9b74-b7fb149572a2.png)

**Comment by JordanPeltier** [comment]
@dougsouza 
Do I have to preprocess the picture before using the frontalize code ? Like face detection, resize or something?

**Comment by dougsouza** [comment]
Yes, but that is already in the frontalization code. It is necessary to detect the face and pass the rectangle of the detection to the landmark detector, if you go through the code you'll see. If you reduce your photo and keep only your face, does it work better? You may also try to resize the whole to a smaller size to see what happens.

**Comment by JordanPeltier** [comment]
I did some preprocessing, here the results : 
![figure_4](https://cloud.githubusercontent.com/assets/13131858/15396732/4610f22c-1dd5-11e6-862c-95aa62814adb.png)
![figure_2](https://cloud.githubusercontent.com/assets/13131858/15396731/4611244a-1dd5-11e6-9c85-f28517ffa1cd.png)
![figure_3](https://cloud.githubusercontent.com/assets/13131858/15396733/461147fe-1dd5-11e6-822f-8a436eefe1f9.png)
![figure_1](https://cloud.githubusercontent.com/assets/13131858/15396734/4624ca04-1dd5-11e6-8b6e-96415a4b1eab.png)

**Comment by JordanPeltier** [comment]
Ok, I tried as well with smaller pictures (the whole smaller) and it works much better ! 
Do you know if there is the same robustess issue in the original matlab code ? 
Thanks a lot

**Comment by dougsouza** [comment]
@Justidan,

I don't. In fact, I haven't used much the matlab code, also, using matlab is a little annoying when using dlib as a landmark detector since dlib runs in python. It is possible to use other detectors though, if you want to try, please refer to the original project page: http://www.openu.ac.il/home/hassner/projects/frontalize/

**Comment by 0x10cxR1** [comment]
The code doesn't contain the crop and resize part. You need to do cropping and resizing according to the paper.
